### DISCUSSION TITLE
Passing the context of machine in invoke

### DISCUSSION THREAD

**Discussion by gaur23rahul** [statelyai/xstate#4943] [Q&A]
Consider the scenario, where I have a state  similar to

```
    send_a_message_W2CuyOyfi: {
      invoke: {
        id: 'send_a_message_W2CuyOyfi',
        src: 'send_message',
       input: {
           'endpoint': 'xxxxxxx',
           'userId': 'xxxxxxx'
       }
        onDone: {
          target: 'ask_a_question_2BkdChpLP',
        },
        onError: {
          target: 'error_handling_w2cuyoyfi',
        },
      },
    },
```

Now suppose I want to send information from context of current machine to the invoked function which is like

```
const send_message = fromPromise(async ({ input: { endpoint, userId } }) => {
    await fetch(`${endpoint}/${userId}`);
   return true;
  }),
```

I want to be able to access data in context of machine in the invoke function. Is that possible and if yes then how. I tried going through docs. In the docs for v5, I didn't see any example/mention of using context in such scenarios.  Serialization is my hard requirement. Any help would be really helpful. Thanks in advance.

**Comment by with-heart** [comment]
You can use the function version of `input` to pass data from `context`. See the second example here: https://stately.ai/docs/input#invoking-actors-with-input

**Reply by gaur23rahul** [reply]
hey @with-heart ,
Thank for reply but serialization is my hard requirement, and if I put function here, I wont be able to store it in database.

**Reply by davidkpiano** [reply]
We're still thinking about the best way to serialize selectors, which will help here.
